src/algebra/lie/basic.lean

```diff
@@ -371,6 +453,58 @@ def inverse (f : M →ₗ⁅R,L⁆ N) (g : N → M)
               ... = ⁅x, g n⁆ : (h₁ _),
   ..linear_map.inverse f.to_linear_map g h₁ h₂ }
 
+instance : has_add (M →ₗ⁅R,L⁆ N) :=
+{ add := λ f g, { map_lie' := by simp, ..((f : M →ₗ[R] N) + (g : M →ₗ[R] N)) }, }
+
+instance : has_sub (M →ₗ⁅R,L⁆ N) :=
+{ sub := λ f g, { map_lie' := by simp, ..((f : M →ₗ[R] N) - (g : M →ₗ[R] N)) }, }
+
+instance : has_neg (M →ₗ⁅R,L⁆ N) :=
+{ neg := λ f, { map_lie' := by simp, ..(-(f : (M →ₗ[R] N))) }, }
+
+@[norm_cast, simp] lemma coe_add (f g : M →ₗ⁅R,L⁆ N) : ⇑(f + g) = f + g := rfl
+
+lemma add_apply (f g : M →ₗ⁅R,L⁆ N) (m : M) : (f + g) m = f m + g m := rfl
+
+@[norm_cast, simp] lemma coe_sub (f g : M →ₗ⁅R,L⁆ N) : ⇑(f - g) = f - g := rfl
+
+lemma sub_apply (f g : M →ₗ⁅R,L⁆ N) (m : M) : (f - g) m = f m - g m := rfl
+
+@[norm_cast, simp] lemma coe_neg (f : M →ₗ⁅R,L⁆ N) : ⇑(-f) = -f := rfl
+
+lemma neg_apply (f : M →ₗ⁅R,L⁆ N) (m : M) : (-f) m = -(f m) := rfl
+
+instance : add_comm_group (M →ₗ⁅R,L⁆ N) :=
+{ zero           := 0,
+  add            := (+),
+  neg            := has_neg.neg,
+  sub            := has_sub.sub,
+  nsmul          := λ n f, { map_lie' := λ x m, by simp, ..(n • (f : M →ₗ[R] N)) },
+  nsmul_zero'    := λ f, by { ext, simp, },
+  nsmul_succ'    := λ n f, by { ext, simp [nat.succ_eq_one_add, add_nsmul], },
+  add_assoc      := λ f g h, by { ext, simp only [add_assoc, add_apply], },
+  zero_add       := λ f, by { ext, simp only [add_apply, zero_add, zero_apply], },
+  add_zero       := λ f, by { ext, simp only [add_zero, add_apply, zero_apply], },
+  sub_eq_add_neg := λ f g, by { ext, simp only [sub_eq_add_neg, sub_apply, add_apply, neg_apply], },
+  add_left_neg   := λ f, by { ext, simp only [add_apply, add_left_neg, neg_apply, zero_apply], },
+  add_comm       := λ f g, by { ext, simp only [add_comm, add_apply], }, }
```

Comment:
I think you might still be able to avoid these proofs with something like
```suggestion
  .. coe_injective.add_comm_group coe rfl rfl rfl rfl }
```
although the `rfl`s may need marginally more proof than that

Reply:
I see the idea here but I can't quite get it to work.

Reply:
Apart from this suggestion, LGTM. @eric-wieser do you think you can make it work?

Reply:
This works:
```suggestion
  ..(coe_injective.add_comm_group _ coe_zero coe_add coe_neg coe_sub :
    add_comm_group (M →ₗ⁅R,L⁆ N)) }
```

Reply:
Nice one!